### PR TITLE
Added a log of departments and employees 

### PR DESCRIPTION
**Pull request by Kostya-Vystorobets**

### CODE REVIEW

File: src/department/department.entity.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   UpdateDateColumn,
 } from "typeorm";
 
 @Entity({ name: "departments" })
 export class DepartmentEntity {
+  getById(
+    departmentId: number
+  ): DepartmentEntity | PromiseLike<DepartmentEntity> {
+    throw new Error("Method not implemented.");
+  }
```
Comment (didevgen) [inline src/department/department.entity.ts:17]: it should not belong to the entity

File: src/department/department.entity.ts [hunk 2]
```diff
@@ -22,4 +29,7 @@ export class DepartmentEntity {
 
   @UpdateDateColumn({ name: "updated_at" })
   updatedAt: Date;
+
+  @OneToMany(() => EmployeeEntity, (employee) => employee.department)
+  employees: EmployeeEntity[];
```
Comment (didevgen) [inline src/department/department.entity.ts:34]: this relation is ManyToMany
one department can belong to many employees and employee can be in multiple departments isn't it?

Comment (Kostya-Vystorobets) [reply src/department/department.entity.ts:34]: One employee in only one department

File: src/migrations/1652801433976-AddRelationsBetweenDepartmentAndEmployees.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class AddRelationsBetweenDepartmentAndEmployees1652801433976 implements MigrationInterface {
+    name = 'AddRelationsBetweenDepartmentAndEmployees1652801433976'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "employees" ("id" SERIAL NOT NULL, "userName" character varying NOT NULL, "firstName" character varying NOT NULL, "lastName" character varying NOT NULL, "created_at" TIMESTAMP NOT NULL DEFAULT now(), "updated_at" TIMESTAMP NOT NULL DEFAULT now(), "departmentId" integer, CONSTRAINT "PK_b9535a98350d5b26e7eb0c26af4" PRIMARY KEY ("id"))`);
```
Comment (didevgen) [inline src/migrations/1652801433976-AddRelationsBetweenDepartmentAndEmployees.ts:7]: i think there needs to be `create table if not exists`

File: src/department/department.entity.ts (hunk 2 above)
Comment (didevgen) [reply src/department/department.entity.ts:34]: okay

**Review by didevgen — APPROVED** [review]

**Pull request merged**